backend/infrahub/core/node/resource_manager/ip_prefix_pool.py

```diff
@@ -95,12 +97,13 @@ async def get_next(self, db: InfrahubDatabase, prefixlen: int) -> IPNetworkType:
                 branch_agnostic=True,
             )
 
-            pool = PrefixPool(resource.prefix.value)  # type: ignore[attr-defined]
+            pool = IPSet([resource.prefix.value])
             for subnet in subnets:
-                pool.reserve(subnet=str(subnet.prefix))
+                pool.remove(addr=str(subnet.prefix))
 
             try:
-                next_available = pool.get(prefixlen=prefixlen)
+                prefix_ver = ipaddress.ip_network(resource.prefix.value).version
+                next_available = get_next_available_prefix(pool=pool, prefix_length=prefixlen, prefix_ver=prefix_ver)
                 return next_available
             except IndexError:
```

Comment:
Given the failing unit tests, I'd guess that they comes from here. It looks like we should catch a `ValueError` instead of an `IndexError`.

Reply:
Ack, updated!